examples/fluids/navierstokes.c

```diff
@@ -526,6 +526,7 @@ static PetscErrorCode TSMonitor_NS(TS ts, PetscInt stepno, PetscReal time,
   ierr = PetscViewerVTKOpen(PetscObjectComm((PetscObject)Q), filepath,
                             FILE_MODE_WRITE, &viewer); CHKERRQ(ierr);
   ierr = VecView(Qloc, viewer); CHKERRQ(ierr);
+  ierr = PetscViewerDestroy(&viewer); CHKERRQ(ierr);
```

Comment:
This is already [in master](https://github.com/CEED/libCEED/blob/4ec104bbb9931294739dc47481cf59de6af2df1d/examples/fluids/navierstokes.c#L454). Wasn't this branch updated?

Reply:
Hmm! Yes, it is updated but I wasn't able to rebase this file with master (especially after #516) so I had to resolve the conflicts manually. I might have missed this one (or maybe more!). I will compare it with master again to see if I have missed other changes or not. Thanks!

Reply:
Would it help to squash/rebase this PR onto current 'master', or do you want to keep individual commits distinct.

Reply:
But, I have been rebasing this branch onto `master` almost after each merge. Do you mean if I squash all changes and then rebase onto master, then I can catch other misses? 
I am not intending to keep all commits distinct but I thought it could be helpful to keep them while we are still reviewing.

Reply:
I'd assumed you had some conflicts if changes from `master` are reproduced here, but in this case, I think @valeriabarra is mistaken; this line fixes a memory leak that still exists in 'master'.

Reply:
It is very possible that I am mistaken, but I see this [in master](https://github.com/CEED/libCEED/blob/master/examples/fluids/navierstokes.c) line 454, before the `if (user->dmviz) {` block. 

Just curious @LeilaGhaffari, were you manually integrating the changes or by using `git merge` or `git rebase`?

Reply:
Line 454 is
```c
  ierr = VecView(Qloc, viewer); CHKERRQ(ierr);
```
That `viewer` isn't destroyed. The variable next appears on line 485:
```c
  ierr = PetscViewerBinaryOpen(user->comm, filepath, FILE_MODE_WRITE, &viewer);
```
which overwrites the pointer, thereby leaking the memory.

Reply:
Yes, it is fixed in master.
@valeriabarra , I wasn't able to rebase it (I think because both of us had touched most parts of the file) so I thought I could skip this only file while I was rebasing and then, integrating the changes manually. I thought, this way, I could keep git history clean which apparently wasn't a good idea.

Reply:
S0rry, I was referring to the wrong line. I meant line 488 on master, where the viewer is destroyed after viewing the global vector, not the local one. 
Anyways, good that's fixed here! Sorry about the confusion!